pyartifactory/objects.py

```diff
@@ -1,684 +1,745 @@
+        """
+        Create user
+        :param user: NewUser object
+        :return: User
+        """
+        username = user.name
+        try:
+            self.get(username)
+            logging.error(f"User {username} already exists")
+            raise UserAlreadyExistsException(f"User {username} already exists")
+        except UserNotFoundException:
+            data = user.dict()
+            data["password"] = user.password.get_secret_value()
+            self._put(f"api/{self._uri}/{username}", json=data)
+            logging.debug(f"User {username} successfully created")
+            return self.get(user.name)
+
+    def get(self, name: str) -> UserResponse:
+        """
+        Read user from artifactory. Fill object if exist
+        :param name: Name of the user to retrieve
+        :return: UserModel
+        """
+        try:
+            r = self._get(f"api/{self._uri}/{name}")
+            return UserResponse(**r.json())
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == 404 or e.response.status_code == 400:
+                logging.error(f"User {name} does not exist")
+                raise UserNotFoundException(f"{name} does not exist")
+            raise ArtifactoryException from e
+
+    def list(self) -> List[SimpleUser]:
+        """
+        Lists all the users
+        :return: UserList
+        """
+        r = self._get(f"api/{self._uri}")
+        logging.debug("List all users successful")
+        return [SimpleUser(**user) for user in r.json()]
+
+    def update(self, user: User) -> UserResponse:
+        """
+        Updates an artifactory user
+        :param user: NewUser object
+        :return: UserModel
+        """
+        username = user.name
+        self.get(username)
+        self._post(f"api/{self._uri}/{username}", json=user.dict())
+        logging.debug(f"User {username} successfully updated")
+        return self.get(username)
+
+    def delete(self, name: str) -> None:
+        """
+        Remove user
+        :param name: Name of the user to delete
+        :return: None
+        """
+        self.get(name)
+        self._delete(f"api/{self._uri}/{name}")
+        logging.debug(f"User {name} successfully deleted")
+
+
+class ArtifactorySecurity(ArtifactoryObject):
+    _uri = "security"
+
+    def __init__(self, artifactory: AuthModel) -> None:
+        super(ArtifactorySecurity, self).__init__(artifactory)
+
+    def get_encrypted_password(self) -> PasswordModel:
+        """
+        Get the encrypted password of the authenticated requestor.
+        :return: str
+        """
+        response = self._get(f"api/{self._uri}/encryptedPassword")
+        logging.debug("Encrypted password successfully delivered")
+        return PasswordModel(**response.json())
+
+    def create_access_token(
+        self,
+        user_name: str,
+        expires_in: int = 3600,
+        refreshable: bool = False,
+        groups: list = None,
```

Comment:
Can this type be even more precise like `List[str]`? Also if it has a default value of `None`, it should be typed as `Optional[List[str]]`.

Reply:
ok done. (I'm still getting used to mypy...)